versioned_docs/version-2.x/advanced/checkout/temporary-shipping-address.mdx

```diff
@@ -0,0 +1,172 @@
+
+```diff title="web/theme/pages/Checkout/stepDefinitions.js"
+const steps = [
+  {
+    renderProgressItem: () =>
+      renderProgressItem("address", "user", messages.address),
+    renderStep: (props) => (
+      <Address
+        checkoutType={props.checkoutState.checkoutType}
+        isEditingBillingAddress={props.checkoutState.isEditingBillingAddress}
+        isEditingShippingAddress={props.checkoutState.isEditingShippingAddress}
+        initialAddress={{
+          billing: props.checkoutState.billingAddress,
+          shipping: props.checkoutState.shippingAddress,
+        }}
+        onChooseAddress={(billing, shipping, email) => {
+          props.checkoutTransaction(() => {
+            props.setEmail(email);
+            props.setBillingAddress(billing);
+            if (shipping) {
+              props.setShippingAddress(shipping);
+            }
+          });
+        }}
+        shouldAskShipping={props.checkoutState.isShippable ?? true}
+      />
+    ),
+    isValid: (checkoutState) => {
+-      const isValidAddress = (address) =>
+-        checkoutState.checkoutType === ACCOUNT_TYPE.GUEST
+-          ? address
+-          : address?.id;
+      const hasGuestInfo =
+        checkoutState.checkoutType !== ACCOUNT_TYPE.GUEST ||
+        checkoutState.email;
+
+      return (
+        isValidAddress(checkoutState.billingAddress) &&
+        (!checkoutState.isShippable ||
+-          isValidAddress(checkoutState.shippingAddress)) &&
+        hasGuestInfo
+      );
+    },
+    isRelevant: () => true,
+    isDisplayable: () => true,
+  },
+```
+
+You must be sure to pass a shipping address without ID to the
+`ChooseShippingMethod` component in the `stepsDefinition.js` file by editing the
+`Address` component that handle address selection.
+
+```jsx title="web/theme/pages/Checkout/stepDefinitions.js"
+<ChooseShippingMethod
+  // highlight-start
+  // On the line below, if an address has an id, it will use id in address book
+  // if not it will use the temporary address
+  shippingAddress={props.checkoutState.shippingAddress}
+  // highlight-end
+  billingAddress={props.checkoutState.billingAddress}
+  setShippingAddress={props.setShippingAddress}
+  setBillingAddress={props.setBillingAddress}
+  initialShipping={{
+    shippingMethod: props.checkoutState.shippingMethod,
+    shippingAdditionalInfo: props.checkoutState.shippingAdditionalInfo,
+  }}
+  onChooseShippingMethod={(method) => {
+    const { additionalData, ...rest } = method;
+    props.checkoutTransaction(() => {
+      props.setShippingMethod(rest);
+      props.setShippingAdditionalInfo(additionalData || {});
+    });
+  }}
+  onChangeShippingAddress={() => props.gotoStepNumber(0)}
+/>
+```
+
+## Nagociable quote checkout
+
+You need to override the `EnhanceAddress` enhancer. And on set checkout shipping
+address, you must add an input to the mutation to specify if you need to save or
+not the address. It's original location is :
```

Comment:
**nit:** I find the sentence a bit harsh to read.

**suggestion:** Rework the sentence.
```suggestion
You will need to override the `EnhanceAddress` enhancer, and add an input to the mutation to specify if the address needs to be save or not. Its original location is :
```